packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxisLayout.selectors.ts

```diff
@@ -0,0 +1,17 @@
+import { createSelector } from '../../utils/selectors';
+import { UseChartCartesianAxisSignature } from './useChartCartesianAxis.types';
+import { ChartState } from '../../models/chart';
+
+export const selectorChartCartesianAxisState = (
+  state: ChartState<[], [UseChartCartesianAxisSignature]>,
```

Comment:
Had to move the `UseChartCartesianAxisSignature` to the second "array" for the types to be correct. Should we have a `OptionalChartState` instead or would it be okay to keep it like this? 😆 

We could have `ChartState|ChartCoreState` too to mark plugins that are always available 🤔

Reply:
For me it's ok to keep it like this. Maybe we should just remove the first option.

We have hooks that use the selectors without guaranteeing that the plugin has been initialized. SO technically all or selectors could be unsure about the existence or not of the state they are looking for in the store.

I already face this multiple time in cases like the pie chart tooltip that complains because the pie do not implement cartesian